### ISSUE TITLE
Attribute value passed to `FactoryBot#create` is being frozen

### ISSUE THREAD

**Issue by krisleech** [thoughtbot/factory_bot#1080]
```ruby
key = 'foobar'

puts key.frozen? # false

FactoryBot.create(:text_field_definition, key: key)

puts key.frozen? # true
```

The first problem is I can't reproduce this on a smaller scale, for example [this gist](https://gist.github.com/krisleech/264737080e6e6de4ae491a9044d01465) does not reproduce the problem :cry:.

In my production code however it seems the attribute value, `"foobar"`, is being frozen during `strategy.result(evaluation).tap(&block)`, in `FactoryBot::Factory#run`:

```ruby
overrides[:key].frozen? # => false
strategy.result(evaluation) # .tap(&block)
overrides[:key].frozen? # => true
```

But I can't for the life of me find exactly where it is happening using a debugger. I also can't see any calls to `#frozen` when grepping the factory_bot code.

In `FactoryBot::Stratgery::Create#result` it seems `evaluation` and `instance` are dup's of the attribute value, `instance.key.object_id` and `evaluation.object.key.object_id` both differing from the `object_id` of the initial attribute value passed in to `FactoryBot.create`. Further adding to my confusion.

I wonder if anyone has any ideas or has experienced this...

Not that my class `TextFieldDefinition` does not freeze any attributes passed to it and as far as I know neither does `ActiveRecord` itself.

* factory_bot 4.8.2
* factory_bot_rails 4.8.2

**Comment by jferris** [comment]
Can you try creating an instance without using FactoryBot?

``` ruby
key = 'foobar'

TextFieldDefinition.create(key: key)

puts key.frozen?
```

**Comment by krisleech** [comment]
`puts key.frozen? # => true` I didn't expect that.

I'm upgrading to Rails 5, going back to Rails 4 I see that `puts key.frozen? # => false`.

After a bit more digging it looks like this might be a Rails 5 issue.

```ruby
KEY = 'foo'
class String; def freeze; raise if self == KEY; self; end
TextFieldDefinition.create(key: KEY)

# active_model/type/immutable_string.rb
# 19: def cast_value(value)
#    20:   result = case value
#    21:            when true then "t"
#    22:            when false then "f"
#    23:            else value.to_s
#    24:            end
#    27: end
```

Unfortunately, once again, I can't reproduce this with a standalone project. ImmutableString does not seem to be used.

I'll dig a bit more.

**Comment by krisleech** [comment]
Yes, this was fixed in Rails 5.1 (I was upgrading to Rails 5.0.1 first as suggested in the guides). Thanks for the tip.
